src/form-elements/FormElementCompliance.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from 'react'
+import clsx from 'clsx'
+
+import FormElementOptions from '../components/FormElementOptions'
+import useFormElementOptions from '../hooks/useFormElementOptions'
+import useBooleanState from '../hooks/useBooleanState'
+import useToggleComplianceChildren from '../hooks/useToggleComplianceChildren'
+import { FormTypes } from '@oneblink/types'
+import OptionButton from './OptionButton'
+import FormElementLabelContainer from '../components/FormElementLabelContainer'
+import ComplianceButton from './ComplianceButton'
+import FormElementFiles, { FilesElementFile } from './FormElementFiles'
+import FormElementTextarea from './FormElementTextarea'
+
+interface Props {
+  id: string
+  element: FormTypes.ComplianceElement
+  value: unknown
+  onChange: (
+    formElement: FormTypes.FormElement,
+    newValue: Value | undefined,
+  ) => unknown
+  onConditionallyShowOption: (
+    choiceElementOption: FormTypes.ChoiceElementOption,
+  ) => boolean
+  displayValidationMessage: boolean
+  validationMessage: string | undefined
+  isEven?: boolean
+}
+
+interface Value {
+  value: unknown
+  notes?: string
+  files?: FilesElementFile[]
+}
+
+const baseElement = {
+  conditionallyShow: false,
+  isDataLookup: false,
+  isElementLookup: false,
+  readOnly: false,
+  required: true,
+  requiresAllConditionallyShowPredicates: false,
+}
+
+const generateNotesElement = (
+  element: FormTypes.ComplianceElement,
+): FormTypes.TextareaElement => ({
+  ...baseElement,
+  id: `${element.id}-notes`,
+  label: 'Notes',
+  name: `${element.name}_notes`,
+  type: 'textarea',
+})
+const generateFilesElement = (
+  element: FormTypes.ComplianceElement,
+): FormTypes.FilesElement => ({
+  ...baseElement,
+  id: `${element.id}-files`,
+  label: 'Media',
+  name: `${element.name}_files`,
+  type: 'files',
+  maxEntries: undefined,
+  minEntries: undefined,
+  restrictFileTypes: false,
+})
+function FormElementCompliance({
+  id,
+  element,
+  value,
+  onChange,
+  onConditionallyShowOption,
+  validationMessage,
+  displayValidationMessage,
+  isEven,
+}: Props) {
+  const typedValue: Value = React.useMemo(() => {
+    if (typeof value === 'object' && !!value) {
+      return value as Value
+    } else {
+      return {
+        value: undefined,
+      }
+    }
+  }, [value])
+
+  const notesElement = React.useMemo(() => generateNotesElement(element), [
+    element,
+  ])
+  const filesElement = React.useMemo(() => generateFilesElement(element), [
+    element,
+  ])
+
+  const handleValueChange = React.useCallback(
+    (fe: FormTypes.FormElement, v: unknown) => {
+      console.log({ v })
```

Comment:
Can we update this to change some context or remove it.

Reply:
accident